tiledb/sm/query/readers/aggregators/aggregate_buffer.cc

```diff
@@ -101,5 +101,35 @@ AggregateBuffer::AggregateBuffer(
           bitmap_data != nullptr ? std::make_optional(bitmap_data) : nullopt) {
 }
 
+AggregateBuffer::AggregateBuffer(
+    const bool var_sized,
+    const bool nullable,
+    const bool count_bitmap,
+    const uint64_t cell_size,
+    const uint64_t min_cell,
+    const uint64_t max_cell,
+    const uint64_t cell_num,
+    ResultTile::TileTuple& tile_tuple,
+    void* bitmap_data)
+    : includes_last_var_cell_(var_sized && max_cell == cell_num)
```

Comment:
Also, we need to remove the manual bookkeeping like [this](https://github.com/TileDB-Inc/TileDB/blob/a0849162272e6b90cd077ef108157696a1dc4752/tiledb/sm/query/readers/aggregators/min_max_aggregator.cc#L330-L335) which exposes internal details to the compute functions. The compute functions should _only_ interact with an interface class that exposes a slicing operator to get the data view by cell id.

Reply:
We've previously discussed this in https://github.com/TileDB-Inc/TileDB/pull/4195#discussion_r1277902381... This will fully go away when we include the last offset inside of the offsets buffer.

Reply:
No. Writing compute operators that know _anything about_ "last offset inside of the offsets buffer" means that we have tied this compute function API far too closely to the underlying data model.

We should be able to change the representation without touching _any code_ in `aggregates/`.

Reply:
Right now, we cannot implement what you want. We know the size of the last element by looking at the var tile size so we always need to special case the last cell. For that reason we need to know if the span includes the last cell. We will fix the offset tiles to include this extra offset at the end of the buffer and then it will be clean. But at the moment we need this...

Reply:
Marking resolved. We will work in parallel to write a var-size container to decouple this API boundary.